crates/core/src/kernel/snapshot/mod.rs

```diff
@@ -738,6 +738,36 @@ mod datafusion {
     }
 }
 
+/// Checks if provided string is enclosed in char_to_remove and extracts the enclosed string.
+/// If the string is not enclosed, it returns the original string.
+fn extract_enclosed_string(string_to_extract: &str, enclosing_char: char) -> &str {
+    if string_to_extract.len() > 1
+        && string_to_extract.starts_with(enclosing_char)
+        && string_to_extract.ends_with(enclosing_char)
+    {
+        &string_to_extract[1..string_to_extract.len() - 1]
+    } else {
+        string_to_extract
+    }
+}
+
+/// Retrieves a specific field from the schema based on the provided field name.
+/// It handles cases where the field name is nested or enclosed in backticks.
+fn get_stats_field<'a>(schema: &'a StructType, stats_field_name: &str) -> Option<&'a StructField> {
```

Comment:
this would require the datafusion flag enabled which idk how everyone would feel for this portion of the code, we could use sqlparser crate directly which datafusion is reexporting there.

Reply:
Yeah that's also an option, didn't consider that tbh! : )